tools/c7n_azure/tests/test_cosmos_db.py

```diff
@@ -149,6 +168,119 @@ def test_offer_collection(self):
         self.assertEqual('Hash', resources[0]['partitionKey']['kind'])
 
 
+class CosmosDBFirewallActionTest(BaseTest):
+
+    @patch('azure.mgmt.cosmosdb.operations.database_accounts_operations.'
+           'DatabaseAccountsOperations.create_or_update')
+    @cassette_name('firewall_action')
+    @arm_template('cosmosdb.json')
+    def test_set_ip_range_filter_append(self, update_mock):
+        p = self.load_policy({
+            'name': 'test-azure-cosmosdb',
+            'resource': 'azure.cosmosdb',
+            'filters': [
+                {'type': 'value',
+                 'key': 'name',
+                 'op': 'eq',
+                 'value_type': 'normalize',
+                 'value': 'cctestcosmosdb'}],
+            'actions': [
+                {'type': 'set-firewall-rules',
+                 'append': True,
+                 'ip-rules': ['0.0.0.0/1', '11.12.13.14', '21.22.23.24']
+                 }
+            ]
+        })
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+        name, args, kwargs = update_mock.mock_calls[0]
+
+        self.assertEqual(resources[0]['resourceGroup'], args[0])
+        self.assertEqual(resources[0]['name'], args[1])
+        self.assertEqual(
+            set('0.0.0.0/1,128.0.0.0/1,11.12.13.14,21.22.23.24'.split(',')),
+            set(kwargs['create_update_parameters']['properties']['ipRangeFilter'].split(',')))
+
+    @patch('azure.mgmt.cosmosdb.operations.database_accounts_operations.'
+           'DatabaseAccountsOperations.create_or_update')
+    @cassette_name('firewall_action')
+    @arm_template('cosmosdb.json')
+    def test_set_ip_range_filter_replace(self, update_mock):
```

Comment:
These tests break the ReplaceOffer one. I'm currently trying to rerecord it, but it's being blocked by firewall rules. Should we create separate test cosmos accounts for firewall rules and offer actions?

Reply:
Probably I left that cosmos instance broken, I am not currently changing them at all in the tests since I patch it.   It is meant to have two rules which allow basically everything as it is provisioned.